deeplake/integrations/mmdet/mmdet_.py

```diff
@@ -0,0 +1,1135 @@
+    ("LTRB", "frac"): pascal_frac_2_coco_pixel,
+    ("CCWH", "pixel"): yolo_pixel_2_coco_pixel,
+    ("CCWH", "frac"): yolo_frac_2_coco_pixel,
+}
+
+
+def convert_to_coco_format(bbox, bbox_format, images):
+    converter = BBOX_FORMAT_TO_COCO_CONVERTER[bbox_format]
+    return converter(bbox, images)
+
+
+class MMDetDataset(TorchDataset):
+    def __init__(
+        self,
+        *args,
+        tensors_dict=None,
+        mode="train",
+        metrics_format="PascalVOC",
+        bbox_info=None,
+        pipeline=None,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.bbox_info = bbox_info
+        self.images = self._get_images(tensors_dict["images_tensor"])
+        self.masks = self._get_masks(
+            tensors_dict.get("masks_tensor", None), shape=self.images[0].shape
+        )
+        self.bboxes = self._get_bboxes(tensors_dict["boxes_tensor"])
+        bbox_format = get_bbox_format(self.bboxes[0], bbox_info)
+        self.labels = self._get_labels(tensors_dict["labels_tensor"])
+        self.iscrowds = self._get_iscrowds(tensors_dict.get("iscrowds"))
+        self.CLASSES = self.get_classes(tensors_dict["labels_tensor"])
+        self.mode = mode
+        self.metrics_format = metrics_format
+        coco_style_bbox = convert_to_coco_format(self.bboxes, bbox_format, self.images)
+
+        if self.metrics_format == "COCO" and self.mode == "val":
+            self.evaluator = mmdet_utils.COCODatasetEvaluater(
+                pipeline,
+                classes=self.CLASSES,
+                hub_dataset=self.dataset,
+                imgs=self.images,
+                masks=self.masks,
+                bboxes=coco_style_bbox,
+                labels=self.labels,
+                iscrowds=self.iscrowds,
+                bbox_format=bbox_format,
+            )
+        else:
+            self.evaluator = None
+
+    def _get_images(self, images_tensor):
+        image_tensor = self.dataset[images_tensor]
+        # image_list = [image.shape for image in image_tensor]
+        return image_tensor
+
+    def _get_masks(self, masks_tensor, shape):
+        if masks_tensor is None:
+            return []
+        ret = self.dataset[masks_tensor].numpy(aslist=True)
+        if self.dataset[masks_tensor].htype == "polygon":
+            ret = map(partial(poly_2_mask, shape=shape), ret)
+        return ret
+
+    def _get_iscrowds(self, iscrowds_tensor):
+        if iscrowds_tensor is not None:
+            return iscrowds_tensor
+
+        if "iscrowds" in self.dataset:
+            always_warn(
+                "Iscrowds was not specified, searching for iscrowds tensor in the dataset."
+            )
+            return self.dataset["iscrowds"].numpy(aslist=True)
+        always_warn("iscrowds tensor was not found, setting its value to 0.")
+        return iscrowds_tensor
+
+    def _get_bboxes(self, boxes_tensor):
+        return self.dataset[boxes_tensor].numpy(aslist=True)
```

Comment:
Maybe fetch_chunks=True? Although I suppose that logic would automatically kick in